runtime/common/src/crowdloan.rs

```diff
@@ -354,101 +354,43 @@ decl_module! {
 		}
 
-		/// Set the deploy data of the funded parachain if not already set. Once set, this cannot
-		/// be changed again.
-		///
-		/// - `origin` must be the fund owner.
-		/// - `index` is the fund index that `origin` owns and whose deploy data will be set.
-		/// - `code_hash` is the hash of the parachain's Wasm validation function.
-		/// - `initial_head_data` is the parachain's initial head data.
-		#[weight = 0]
-		fn fix_deploy_data(origin,
-			#[compact] index: FundIndex,
-			code_hash: T::Hash,
-			code_size: u32,
-			initial_head_data: HeadData,
-		) {
-			let who = ensure_signed(origin)?;
-
-			let mut fund = Self::funds(index).ok_or(Error::<T>::InvalidFundIndex)?;
-			ensure!(fund.owner == who, Error::<T>::InvalidOrigin); // must be fund owner
-			ensure!(fund.deploy_data.is_none(), Error::<T>::ExistingDeployData);
-
-			fund.deploy_data = Some(DeployData { code_hash, code_size, initial_head_data });
-
-			<Funds<T>>::insert(index, &fund);
-
-			Self::deposit_event(RawEvent::DeployDataFixed(index));
-		}
-
-		/// Complete onboarding process for a winning parachain fund. This can be called once by
-		/// any origin once a fund wins a slot and the fund has set its deploy data (using
-		/// `fix_deploy_data`).
-		///
-		/// - `index` is the fund index that `origin` owns and whose deploy data will be set.
-		/// - `para_id` is the parachain index that this fund won.
-		#[weight = 0]
-		fn onboard(origin,
-			#[compact] index: FundIndex,
-			#[compact] para_id: ParaId
-		) {
-			ensure_signed(origin)?;
-
-			let mut fund = Self::funds(index).ok_or(Error::<T>::InvalidFundIndex)?;
-			let DeployData { code_hash, code_size, initial_head_data }
-				= fund.clone().deploy_data.ok_or(Error::<T>::UnsetDeployData)?;
-			ensure!(fund.parachain.is_none(), Error::<T>::AlreadyOnboard);
-			fund.parachain = Some(para_id);
-
-			let fund_origin = frame_system::RawOrigin::Signed(Self::fund_account_id(index)).into();
-			<slots::Module<T>>::fix_deploy_data(
-				fund_origin,
-				index,
-				para_id,
-				code_hash,
-				code_size,
-				initial_head_data,
-			)?;
-
-			<Funds<T>>::insert(index, &fund);
-
-			Self::deposit_event(RawEvent::Onboarded(index, para_id));
-		}
-
 		/// Note that a successful fund has lost its parachain slot, and place it into retirement.
 		#[weight = 0]
-		fn begin_retirement(origin, #[compact] index: FundIndex) {
+		fn begin_retirement(origin, #[compact] index: ParaId) {
 			ensure_signed(origin)?;
 
-			let mut fund = Self::funds(index).ok_or(Error::<T>::InvalidFundIndex)?;
-			let parachain_id = fund.parachain.take().ok_or(Error::<T>::NotParachain)?;
+			let mut fund = Self::funds(index).ok_or(Error::<T>::InvalidParaId)?;
+			let parachain_id = fund.parachain;
+
 			// No deposit information implies the parachain was off-boarded
-			ensure!(<slots::Module<T>>::deposits(parachain_id).len() == 0, Error::<T>::ParaHasDeposit);
+			ensure!(!T::Registrar::is_registered(parachain_id), Error::<T>::ParaAlive);
```

Comment:
This is not a valid check. `is_registered` can return true for a parathread, which should still allow a fund to be retired.

I think we need to also import the `Leaser` trait, and check `deposit_held` is zero